x-pack/platform/plugins/shared/lens/public/editor_frame_service/editor_frame/config_panel/esql_editor.tsx

```diff
@@ -108,12 +107,6 @@ export function ESQLEditor({
 
   const previousAdapters = useRef<Partial<DefaultInspectorAdapters> | undefined>(lensAdapters);
 
-  const esqlVariables = useStateFromPublishingSubject(
-    isApiESQLVariablesCompatible(parentApi)
-      ? parentApi?.esqlVariables$
-      : new BehaviorSubject(undefined)
-  );
-
```

Comment:
You can actually keep this and avoiding passing the esqlVariables from the props. It works correctly with the previous code here.
Seems that only the `layer_panel` changes and the consequent update in the `dimension_editor` are required to make this works.

Reply:
I did it like that to avoid code duplication but happy to do it as you propose

Reply:
Done https://github.com/elastic/kibana/pull/221083/commits/ef6f446e027b3ef532d04de1c6936b0a12cbb4a7